### PR TITLE
Timeline of loadgen stats

### PR DESCRIPTION
**Pull request by mhofman**
This PR provides the plumbing to generate a CSV file containing a small subset of the loadgen stats across multiple revisions.

The file is currently generated locally and manually imported into a google spreadsheet. I still want to figure out a better way, by uploading the results automatically somewhere, and make them available to a hosted dashboard, possibly through a "revision completion" hook for the background script.

I generated a preliminary corpus of data on my own machine, with the results available [here](https://docs.google.com/spreadsheets/d/174eYWAlunqVm0ENXfJWuFBNvIsi6UPX4KqPp4SFFykI/edit#gid=1343630455), using the following config: `--no-stage.save-storage --monitor-interval=1 --stage.duration=10 --stage.loadgen.vault.interval=30 --stage.loadgen.amm.interval=30 --stage.loadgen.amm.wait=15 --stage.loadgen.vault.limit=2 --stage.loadgen.amm.limit=2`

A longer version is currently being generated on the benchmark machine for the same revisions with the following config: `--no-stage.save-storage --monitor-interval=1 --stage.duration=30 --stage.loadgen.vault.interval=60 --stage.loadgen.amm.interval=60 --stage.loadgen.amm.wait=30`

I'd ultimately like to run the default 24h test in the background on a subset of past revisions to backfill the data, but I'm not sure which revisions would be the best candidates yet.

Stack on top of compatibility fixes (#37 / #44) and stats work (#36).

As always, best reviewed commit-by-commit.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -150,12 +150,13 @@ ENV IS_DOCKER=true
 ENV SDK_SRC=/src
 ENV OUTPUT_DIR=/out
 ENV SDK_REVISION=
+Env SDK_REPO=
```
Comment (michaelfig) [inline Dockerfile:153]: This is the first time I've seen a non-capitalised Docker command.  Interesting.

File: Dockerfile [hunk 2]
```diff
@@ -150,12 +150,13 @@ ENV IS_DOCKER=true
...
 ENV SDK_BUILD=0
 
 WORKDIR /app
 COPY --chown=$USER_UID:$USER_GID . .
 
-RUN mkdir -p $SDK_SRC $OUTPUT_DIR && chown $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /app
+RUN mkdir -p $SDK_SRC $OUTPUT_DIR /home/node/.cache/yarn /go/pkg/mod && chown -R $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /home/node/.cache /go && chown $USER_UID:$USER_GID /app
```
Comment (michaelfig) [inline Dockerfile:159]: Please split long lines.

```suggestion
RUN mkdir -p $SDK_SRC $OUTPUT_DIR /home/node/.cache/yarn /go/pkg/mod && \
  chown -R $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /home/node/.cache /go && \
  chown $USER_UID:$USER_GID /app
```

File: scripts/common-queue.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      [ ! -d "${OUTPUT_DIR}" ] && break
+      [ "x$SDK_REVISION" != "x$PREV_SDK_REVISION" ] && continue
+      sleep 60 &
+      SLEEP_PID=$!
+      wait $SLEEP_PID
+      SLEEP_PID=
+    done
+    echo "processing ${SDK_REVISION}"
+    mkdir "${OUTPUT_DIR}"
+    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod -v loadgen-yarn-cache:/home/node/.cache/yarn -v "$(pwd)/${OUTPUT_DIR}:/out" -e SDK_REVISION=${SDK_REVISION} -e SDK_REPO=${SDK_REPO} --name "${OUTPUT_DIR}" loadgen-runner --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```
Comment (michaelfig) [inline scripts/common-queue.sh:37]:
```suggestion
    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod \
      -v loadgen-yarn-cache:/home/node/.cache/yarn \
      -v "$(pwd)/${OUTPUT_DIR}:/out" \
      -e SDK_REVISION=${SDK_REVISION} \
      -e SDK_REPO=${SDK_REPO} \
      --name "${OUTPUT_DIR}" \
      loadgen-runner \
      --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```

File: scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Requires a docker image named `loadgen-runner`. Extra arguments are passed to the container.
+
+### Daily perf service
+
+`loadgen-daily-perf.service` is an example of systemd service file to automatically run the daily-perf loadgen testing.
+
+### Manual perf example
+
+```sh
+cat << EOF | /path/to/testnet-load-generator/scripts/run-manual.sh - --no-stage.save-storage --stage.duration=30 --monitor-interval=1 --stage.loadgen.vault.interval=60 --stage.loadgen.amm.interval=60 --stage.loadgen.amm.wait=30
```
Comment (michaelfig) [inline scripts/README.md:16]:
```suggestion
cat << EOF | /path/to/testnet-load-generator/scripts/run-manual.sh - \
  --no-stage.save-storage \
  --stage.duration=30 \
  --monitor-interval=1 \
  --stage.loadgen.vault.interval=60 \
  --stage.loadgen.amm.interval=60 \
  --stage.loadgen.amm.wait=30
```

File: scripts/perf_to_stats_csv.jq [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env -S jq -Mf -s -r
+
+[ 
+  .[] |
+  select(has("stats")) |
+  .stats |
+  {
+    sdkRevision: (.metadata.testData.sdkRevision | tostring), 
+    sdkCommitTime: (.metadata.testData.sdkCommitTime | strftime("%Y-%m-%d %H:%M:%S")),
```
Comment (michaelfig) [inline scripts/perf_to_stats_csv.jq:9]: What timezone is the sdkCommitTime measured in?  I'd prefer UTC, if possible.

**Review by michaelfig — APPROVED** [review]
LGTM!  I have just a few suggestions, but you're welcome to rebut them if you don't find them useful.

File: scripts/perf_to_stats_csv.jq (hunk 5 above)
Comment (mhofman) [reply scripts/perf_to_stats_csv.jq:9]: This is plumbed from the `git show` command producing a Unix timestamp:
https://github.com/Agoric/testnet-load-generator/blob/d8c3e286349ca43d7ac64aac1a3bfb1bf7109295/start.sh#L37

And according to [`jq`'s manual](https://stedolan.github.io/jq/manual/#Dates), `strftime` formats using UTC/GMT: 
> The `strftime(fmt)` builtin formats a time (GMT) with the given format. The `strflocaltime` does the same, but using the local timezone setting.

**Review by mhofman — COMMENTED** [review]
Thanks for catching those. I agree, splitting long lines is more readable. I need a prettier for bash, I've become to lazy to format my lines manually.

**Review by warner — APPROVED** [review]
nice

File: Dockerfile (hunk 1 above)
Comment (mhofman) [reply Dockerfile]: Resolved in 41242a65e6bbd5660c7df1262e05ec069b09e022

File: Dockerfile (hunk 2 above)
Comment (mhofman) [reply Dockerfile]: Resolved in 6287b8096ebd48bb08671feb6c24c422e917dd3f

File: scripts/common-queue.sh (hunk 3 above)
Comment (mhofman) [reply scripts/common-queue.sh]: Resolved in 6287b8096ebd48bb08671feb6c24c422e917dd3f

File: scripts/README.md (hunk 4 above)
Comment (mhofman) [reply scripts/README.md]: Resolved in f0b5f1a8953212b6427cbb93c9367a7f83be9d33

**Comment by mhofman** [comment]
@michaelfig since your initial review I addressed your feedback and added d30dea34b014f0fbd817bf6fcd5acdcede5652e1 and 8717013aaa8b639cfc512fe1a9fbfd2a89328efd.

Since those are minor changes, and @warner approved the PR following them, I will squash the feedback commits, and merge.

**Comment by mhofman** [comment]
Scripts to generate a CSV file containing a small subset of the loadgen stats across multiple revisions.
Increase default loadgen load

**Pull request merged**